homeassistant/components/nobo_hub/strings.json

```diff
@@ -0,0 +1,75 @@
+{
+  "config": {
+    "step": {
+      "manual": {
+        "description": "Failed to discover the Nobø Ecohub on your local network. If your hub is on another network, you can still connect to it by entering the complete serial number (12 digits) and its IP address.",
+        "data": {
+          "serial": "Serial number (12 digits)",
+          "ip_address": "[%key:common::config_flow::data::ip%]"
+        }
+      },
+      "user": {
+        "description": "Select Nobø Ecohub to configure. To connect to the hub, you need to enter the last 3 digits of the hub's serial number.\n\nTo avoid using discovery when connecting, you can tell Home Assistant to remember the IP address. Only use this if your hub has assigned a static IP address.",
+        "data": {
+          "device": "Discovered hubs",
+          "manual": "Configure undiscovered hub",
+          "serial_suffix": "Serial number suffix (3 digits)",
+          "store_ip": "Remember the IP address"
```

Comment:
Hmm - that will add some complexity to the initialisation of the integration. I'll look into it.

Reply:
It's not straight forward nor really necessary to implement periodic discovery. The discovery protocol only replies the serial prefix. To verify the serial suffix, it's necessary to connect to the hub, but the hub only supports 2 concurrent TCP/IP connections, one of these will be in use by the integration, and the other may be in use from e.g. a mobile app. If the IP address is the same, we can  be sure that it is the same hub, but in theory (however unlikely) there may be more than one hub on the local network, and they may have the the same serial number prefix. This corner case was the what "Remember the IP address" was intended to fix.

I agree the choice is a bit confusing, and the above corner case can also be supported by selecting the manual step, so I removed the check box for `store_ip`. The only drawback is that the user then must enter the entire serial number and the IP address, but I guess this rarely will be necessary.

The library itself does reconnect automatically, and if the hub has gotten a new IP address (e.g. from the local DHCP server), it will try to re-discover if the hub was auto-discovered in the first place.

I added always storing the IP address, and in addition a flag to say of the hub was discovered or not.